cranelift/codegen/src/isa/x64/inst.isle

```diff
@@ -3274,6 +3274,13 @@
       ;; Same flags as `UnorderedOrLessThanOrEqual`.
       (FcmpCondResult.Condition (x64_ucomis a b) (CC.BE)))
 
+;;;; Type Guards ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+;; A type guard for matching 64-bit sized ints, bools, or references but
+;; explicitly ruling out 32-bit references.
+(decl ty_int_bool_or_ref () Type)
+(extern extractor ty_int_bool_or_ref ty_int_bool_or_ref)
```

Comment:
Nitpick: I feel like the name could have 64 in it somewhere since this explicitly doesn't handle 32-bit ints/bools/refs.

Reply:
Wait looking at the implementation, this accepts 32-bit ints/bools, jsut not 32-bit refs. Which I think is fine, but I think means the comment here is misleading.

Reply:
The function I ported over seemed to accept a pretty wide range of types, definitely open to better naming suggestions :)